src/routes/transactions/mappers/transfers/swap-transfer-info.mapper.spec.ts

```diff
@@ -0,0 +1,399 @@
+} as jest.MockedObjectDeep<SwapOrderHelper>);
+
+const mockSwapsRepository = jest.mocked({
+  getOrders: jest.fn(),
+} as jest.MockedObjectDeep<ISwapsRepository>);
+
+describe('SwapTransferInfoMapper', () => {
+  let target: SwapTransferInfoMapper;
+
+  const GPv2SettlementAddress = '0x9008D19f58AAbD9eD0D60971565AA8510560ab41';
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+
+    target = new SwapTransferInfoMapper(
+      mockSwapOrderHelper,
+      mockSwapsRepository,
+    );
+  });
+
+  it('it returns null if nether the sender and recipient are from the GPv2Settlement contract', async () => {
+    const sender = addressInfoBuilder().build();
+    const recipient = addressInfoBuilder().build();
+    const direction = faker.helpers.arrayElement(
+      Object.values(TransferDirection),
+    );
+    const chainId = faker.string.numeric();
+    const safeAddress = getAddress(faker.finance.ethereumAddress());
+    const domainTransfer = erc20TransferBuilder()
+      .with('from', getAddress(sender.value))
+      .with('to', getAddress(recipient.value))
+      .build();
+    const token = tokenBuilder()
+      .with('address', domainTransfer.tokenAddress)
+      .build();
+    const transferInfo = new Erc20Transfer(
+      token.address,
+      domainTransfer.value,
+      token.name,
+      token.symbol,
+      token.logoUri,
+      token.decimals,
+      token.trusted,
+    );
+    const order = orderBuilder().with('from', getAddress(sender.value)).build();
+    mockSwapsRepository.getOrders.mockResolvedValue([order]);
+
+    const actual = await target.mapSwapTransferInfo({
+      sender,
+      recipient,
+      direction,
+      chainId,
+      safeAddress,
+      transferInfo,
+      domainTransfer,
+    });
+
+    expect(actual).toBe(null);
+  });
+
+  it('maps the SwapTransferTransactionInfo if the sender is the GPv2Settlement contract', async () => {
+    const sender = addressInfoBuilder()
+      .with('value', GPv2SettlementAddress)
+      .build();
+    const recipient = addressInfoBuilder().build();
+    const chainId = faker.string.numeric();
+    const safeAddress = getAddress(faker.finance.ethereumAddress());
+    const direction = getTransferDirection(
+      safeAddress,
+      sender.value,
+      recipient.value,
+    );
+    const domainTransfer = erc20TransferBuilder()
+      .with('from', getAddress(sender.value))
+      .with('to', getAddress(recipient.value))
+      .build();
+    const token = tokenBuilder()
+      .with('address', domainTransfer.tokenAddress)
+      .build();
+    const transferInfo = new Erc20Transfer(
+      token.address,
+      domainTransfer.value,
+      token.name,
+      token.symbol,
+      token.logoUri,
+      token.decimals,
+      token.trusted,
+    );
+    const order = orderBuilder()
+      .with('from', getAddress(sender.value))
+      .with('owner', safeAddress)
+      .with('buyToken', token.address)
+      .with('executedBuyAmount', BigInt(domainTransfer.value))
```

Comment:
In these tests, I've just adjusted the `Order` payloads to match the logic implemented in `SwapTransferInfoMapper.findOrderByTransfer` not to filter out the order, i.e.: `executedBuyAmount` needs to match the transfer value, and `buyToken` address needs to match the token address. I'm assuming this logic is correct. cc. @iamacook @compojoom